### ISSUE TITLE
Modifying response after `.toJSON()`

### ISSUE THREAD

**Issue by lwansbrough** [balderdashy/sails#2176]
I use `.toJSON(...)` in my models to define how they should appear in JSON responses - however I'm running into some trouble.

It seems that I cannot modify this object after it has been created. This is a problem because in many cases I'll have a property that I do not wish to be available under most circumstances, however there are a few cases where I do need to expose that property.

One example is a `User`'s email - I don't want it to show up in the `/users` list, but I want it to show up when I query `/me`.

The current code I have for `/me` is something like this:

    User.findOne({ id: req.user.id })
            .exec(function(err, user) {
                if(err) return res.negotiate(err);
                if(!user) return res.notFound();
                Alias.find({ owner: user.id }).populate('platform').exec(function(err, aliases) {
                  if(err) return res.serverError(err);
                  user.aliases = aliases;
                  var json = user.toJSON();
                  json.settings.passwordDetails = user.passwordDetails;
                  json.email = user.email;
                  res.json(json);
                });
            });

I would expect that the JSON response should include an `email` property, however it is removed. I checked to make sure that the `json.email` property does in fact exist, and it does if I `console.log` the `json` object. What gives? (Note: The property `json.settings.passwordDetails` DOES work as expected (`settings` is a property of type `json` on `User`) - which I don't fully understand though I have a few ideas.)

**Comment by CWyrtzen** [comment]
Is this resolved?

**Comment by sailsbot** [comment]
Thanks for posting, @lwansbrough.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
